MediaGalleryRenditions/Model/RenditionsImageManagement.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryRenditions\Model;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\Framework\Filesystem\Io\File;
+
+class RenditionsImageManagement
+{
+    /**
+     * @var Filesystem\Directory\WriteInterface
+     */
+    private $_directory;
+
+    /**
+     * @var File|mixed|null
+     */
+    private $ioFile;
+
+    /**
+     * @var DriverInterface|mixed|null
+     */
+    private $file;
+
+    private const RENDITIONS_DIRECTORY_NAME = '.renditions';
+
+    /**
+     * @param Filesystem $filesystem
+     * @param File|null $ioFile
+     * @param DriverInterface|null $file
+     * @throws FileSystemException
+     */
+    public function __construct(
+        Filesystem $filesystem,
+        File $ioFile = null,
+        DriverInterface $file = null
+    ) {
+        $this->_directory = $filesystem->getDirectoryWrite(DirectoryList::MEDIA);
+        $this->ioFile = $ioFile ?: ObjectManager::getInstance()->get(File::class);
+        $this->file = $file ?: ObjectManager::getInstance()->get(Filesystem\Driver\File::class);
+    }
+
+    /**
+     * Returns Rendition image path
+     *
+     * @param string $path
+     * @return string
+     * @throws LocalizedException
+     * @throws FileSystemException
+     */
+    public function execute(string $path) :string
```

Comment:
I think it would be handier to provide ability to get rendition path based on asset id
```suggestion
    public function execute(int $assetId): string
```

Reply:
@sivaschenko ... I think we can get the assets by paths using `Magento\MediaGalleryApi\Api\GetAssetsByPathsInterface` so instead of passing assetId we can pass the `Magento\MediaGalleryApi\Api\Data\AssetInterface` as parameter. Let me know your thoughts

Reply:
If it's more comfortable for the clients to use the path as an argument - that's ok with me